### ISSUE TITLE
RFC: Pulldasher: Show number of lines changed on PRs

### ISSUE THREAD

**Issue by davidrans** [iFixit/pulldasher#225]
@andyg0808 commented on [Tue Oct 12 2021](https://github.com/iFixit/ifixit/issues/40208)

## Problem
We don't have a quick way to see which PRs are likely to be easiest to CR.

## Do This
Add an indicator for number of lines changed to Pulldasher.

## Background
Nerd Herd's brought this up twice in retros now.

CC: @Annika29 

---

@davidrans commented on [Tue Oct 12 2021](https://github.com/iFixit/ifixit/issues/40208#issuecomment-941294293)

Some discussion happened here as well: https://ifixit.slack.com/archives/C90126AN4/p1628184142080500

---

@djmetzle commented on [Tue Oct 12 2021](https://github.com/iFixit/ifixit/issues/40208#issuecomment-941308854)

If the intent is to use LoC changed as a proxy for complexity, lets craft some arbitrary metric like:
```
log_10(lines_added) + log_2(lines_deleted)
```

Or the likes? That way we're not encouraging gaming LoC, just discouraging very large PRs?

---

@davidrans commented on [Tue Oct 12 2021](https://github.com/iFixit/ifixit/issues/40208#issuecomment-941458336)

How about median LoC per commit?

---

@GrantMatejka commented on [Wed Oct 13 2021](https://github.com/iFixit/ifixit/issues/40208#issuecomment-942417911)

I like the idea of this feature, and even though LoC is a notoriously bad metric, I think if we use it _innocently_ and bucket<sup>1</sup> on some convoluted calculation it would work fine. I think an issue like this is very open to bikeshedding so fwmioiw I'm fine trying anything to see if it works/is helpful

<sup>1</sup>bucket as in a pull is either 'long' or not, so on pulldasher maybe a long pulls title is yellow or something but all 'short' pulls don't have any indicator idk

---

@djmetzle commented on [Fri Oct 29 2021](https://github.com/iFixit/ifixit/issues/40208#issuecomment-954918372)

Should we migrate this to the pulldasher repo? Or was the intent here to gather consensus? 

---

@davidrans commented on [Fri Oct 29 2021](https://github.com/iFixit/ifixit/issues/40208#issuecomment-954932738)

Copying my comment from the slack thread here:

I like it. I've been opposed to it in the past, cause I didn't want people to skip long pulls, but if you don't have time to CR a super long pull it's nice to know before you click. And I also think it'll be cool if it results in smaller pulls getting CR'd quicker. It's frustrating when you work hard to break something up into small, readable pulls and it still takes a long time to get CR.

I already added lines changed, and # of commits to the pulldasher DB so it should be pretty easy

---

@danielbeardsley commented on [Fri Oct 29 2021](https://github.com/iFixit/ifixit/issues/40208#issuecomment-954937542)

> I already added lines changed, and # of commits to the pulldasher DB so it should be pretty easy

Oh nice!

---

@andyg0808 commented on [Mon Nov 01 2021](https://github.com/iFixit/ifixit/issues/40208#issuecomment-956570443)

@djmetzle Probably reasonable to migrate to the Pulldasher repo.

---

@djmetzle commented on [Tue Nov 02 2021](https://github.com/iFixit/ifixit/issues/40208#issuecomment-958174937)

@andyg0808 
> @djmetzle Probably reasonable to migrate to the Pulldasher repo.

![image](https://user-images.githubusercontent.com/6876047/139953246-fad72949-abb6-4858-b320-12a7cd026746.png)

Doesn't seem to want to let me. Maybe someone else can give it a try?

**Comment by djmetzle** [comment]
Awesome. Thanks @davidrans

**Comment by jordycosta** [comment]
Currently we show the pull author whenever hovering over the title of the pull request

![Ikgt09GJj3](https://user-images.githubusercontent.com/95656772/225446676-ffa53710-5103-4cdd-a4ca-82572592293b.png)

---

I was thinking that instead we could show the lines changed here instead, which would probably take up the least UI space
> Note: The author title is still shown when hovering on the pull card avatar

![additions_deletions](https://user-images.githubusercontent.com/95656772/225446812-8513dc33-f9b0-49d3-987c-9d9b7b1ed0ab.png)

@danielbeardsley @davidrans  thoughts?

**Comment by danielbeardsley** [comment]
This seems like the best argument for having this information on the dashboard:
> I've been opposed to it in the past, cause I didn't want people to skip long pulls, but if you don't have time to CR a super long pull it's nice to know before you click. And I also think it'll be cool if it results in smaller pulls getting CR'd quicker. It's frustrating when you work hard to break something up into small, readable pulls and it still takes a long time to get CR.

I'm a bit more interested in showing it without hovering. Maybe a small bar who's length corresponds to the number of lines. Maybe even a small green + red bar (for lines added / removed). It could live under the age dots... or on the first right above the refresh button?

**Comment by jordycosta** [comment]
@danielbeardsley Here's what it would look like below the pull request age counter

![Screenshot 2023-03-21 104534](https://user-images.githubusercontent.com/95656772/226696840-ccd3bcd7-9d76-4911-9a6b-23bb96a2227d.jpg)

**Comment by danielbeardsley** [comment]
I was thinking more like this: 
![image](https://user-images.githubusercontent.com/26855/226706878-234fbb67-742d-4bc6-b6cf-694561d1d6d5.png)

Thoughts?

**Comment by jordycosta** [comment]
@danielbeardsley That looks good to me, another thought I had would to have this only for CR column cards

**Comment by davidrans** [comment]
I agree that something visible without hovering is ideal. But, I have no better ideas about the UI

**Comment by jordycosta** [comment]
Here's another simple implementation:

![code](https://user-images.githubusercontent.com/95656772/226740388-3197fad1-c7e6-400b-b8d9-494b23f07b9f.jpg)
